packages/core/contracts/optimistic-asserter/implementation/escalation-manager/FullPolicyEscalationManager.sol

```diff
@@ -0,0 +1,201 @@
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./BaseEscalationManager.sol";
+import "../../interfaces/OptimisticAsserterInterface.sol";
+
+/**
+ * @title The FullPolicyEscalationManager enables the owner to configure all policy parameters and store the arbitration
+ * resolutions for the Escalation Manager. Optionally, assertion blocking can be enabled using a whitelist of
+ * assertingCallers or assertingCallers and asserters. On the other hand, it enables the determination of whether to
+ * arbitrate via the escalation manager as opposed to the DVM, whether to disregard the resolution of a potential
+ * dispute arbitrated by the Oracle, and whether to restrict who can register disputes via a whitelistedDisputeCallers list.
+ * @dev If nothing is configured using the setters and configureEscalationManager method upon deployment, the
+ * FullPolicyEscalationManager will return a default policy with all values set to false.
+ */
+contract FullPolicyEscalationManager is BaseEscalationManager, Ownable {
+    enum AssertionBlockMode {
+        None, // No assertions are blocked
+        BlockByAssertingCallerAndAsserter, // Assertion are block by asserting caller and asserter pair
+        BlockByAssertingCaller // Assertions are blocked by asserting caller
+    }
```

Comment:
The reason for not having 2 separate booleans is that its not secure to filter by asserters while asserting callers are not checked. This is due to the fact that `asserter` can be provided as any address in `assertTruth`

Reply:
That makes sense! What about two booleans and enforcing the rule by reverting if the owner tries to configure it that way? I just found the code to do the checking a bit hard to parse with the enum setup, I also think enums can be a bit more annoying than bools for users.

Reply:
agree the alternative with booleans would be to enforce this in `configureEscalationManager` so that having `blockByAndAsserter` true requires also `blockByAssertingCaller` being true.

Reply:
Yep, exactly.